app/src/util/pdf_utils.py

```diff
@@ -9,6 +8,41 @@
 from pdfminer.psparser import PSLiteral
 
 
+def as_pdf_doc(pdf: BinaryIO | PDFDocument) -> PDFDocument:
+    if isinstance(pdf, PDFDocument):
+        return pdf
+    else:
+        return PDFDocument(PDFParser(pdf))
+
+
+def get_pdf_info(pdf: BinaryIO | PDFDocument, count_pages: bool = False) -> dict[str, Any]:
```

Comment:
Could we introduce a dataclass instead of `dict[str, Any]` so the typing is safer?

Reply:
Sure!